src/util/pelz_io.c

```diff
@@ -280,3 +287,85 @@ int decodeBase64Data(unsigned char *base64_data, size_t base64_data_size, unsign
   BIO_free_all(bio64);
   return (0);
 }
+
+int write_to_pipe(char *msg)
+{
+  int fd;
+  int ret;
+
+  if (file_check((char*) PELZFIFO))
+  {
+    pelz_log(LOG_ERR, "Pipe not found");
+    return 1;
+  }
+
+  fd = open(PELZFIFO, O_WRONLY);
+  ret = write(fd, msg, strlen(msg)+1);
+  close(fd);
+  if (ret == -1)
+  {
+    pelz_log(LOG_ERR, "Error writing to pipe");
+    return 1;
+  }
+  return 0;
+}
+
+int read_pipe(char *msg)
+{
+  int ret;
+  int len;
+  char opt;
+  charbuf key_id;
+
+  if (memcmp(msg, "pelz -", 6) == 0)
+  {
+    opt = msg[6];
+    pelz_log(LOG_DEBUG, "Pipe message: %d, %c, %s", strlen(msg), opt,  msg);
+    switch (opt)
+    {
+    case 't':
+      pthread_mutex_lock(&lock);
+      key_table_destroy(eid, &ret);
+      if (ret)
+      {
+        pelz_log(LOG_ERR, "Key Table Destroy Failure");
+        return (1);
+      }
+      pelz_log(LOG_INFO, "Key Table Destroyed");
+      key_table_init(eid, &ret);
+      if (ret)
+      {
+        pelz_log(LOG_ERR, "Key Table Init Failure");
+        return (1);
+      }
+      pthread_mutex_unlock(&lock);
+      pelz_log(LOG_INFO, "Key Table Re-Initialized");
+      return 0;
+    case 'w':
+      len = strcspn(msg, "\n");
+      key_id = new_charbuf(len - 7);
+      memcpy(key_id.chars, &msg[8], (key_id.len - 1));
+      pthread_mutex_lock(&lock);
+      key_table_delete(eid, &ret, key_id);
+      pthread_mutex_unlock(&lock);
+      if(ret)
+        pelz_log(LOG_ERR, "Delete Key ID from Key Table Failure: %.*s", (int) key_id.len, key_id.chars);
+      else
+        pelz_log(LOG_INFO, "Delete Key ID form Key Table: %.*s", (int) key_id.len, key_id.chars);
+      return 0;
+    case 'e':
+      if (unlink(PELZFIFO) == 0)
+        pelz_log(LOG_INFO, "Pipe deleted successfully");
+      else
+        pelz_log(LOG_INFO, "Failed to delete the pipe");
+      return 1;
+    default:
+      pelz_log(LOG_ERR, "Pipe command invalid");
+      return 0;
+    }
+  }
+  else
+    pelz_log(LOG_ERR, "Pipe command invalid");
```

Comment:
Should the invalid command get logged here?

Reply:
This invalid command could be anything so is not logged.

Reply:
Can we at least log like the first 10 or so characters? Maybe use ```strnlen``` to check if the message is at least 10 characters long, if its shorter log the whole message, if it's longer just the first 10 characters.

Reply:
Added a 10 character log of the invalid message over the pipe.

Reply:
I'm going to be a stickler here for ```strnlen``` over ```strlen``` on this one, since otherwise there's no promise you hit the end of the string in any reasonable length of time.

Reply:
The change has been made.